server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -302,7 +303,11 @@ private RestResponse getDetailResponse(NettyRestResponse.Builder responseBuilder
       InfinispanQueryStatisticsInfo.IndexStatistics indexStatistics = getIndexStatistics(cache);
       boolean rehashInProgress = distributionManager != null && distributionManager.isRehashInProgress();
       boolean indexingInProgress = indexStatistics != null && indexStatistics.getReindexing();
-
+      boolean indexed =  configuration.indexing().enabled();
+      ContentTypeConfiguration valueDataType = cache.getCacheConfiguration().encoding().valueDataType();
+      boolean queryable = indexed || (valueDataType != null && valueDataType.mediaType()!= null &&
```

Comment:
It's wrong to assume a cache without an explicit encoding configuration is not queryable, because depending on the storage type it will either use `APPLICATION_OBJECT` or the user marshaller's media type (which is `APPLICATION_PROTOSTREAM` with the default serialization configuration).

You should use `cache.getValueDataConversion().getStorageMediaType()` to get the actual media type, but I'd much rather make this a method in `query-core`

Reply:
@danberindei You are right, I will send a follow up PR to have a method in StorageManager to verify the runtime storage

Reply:
This will also come handy for many other places that need this info

Reply:
The PR already exists it's https://github.com/infinispan/infinispan/pull/8389  :)